src/viewer/CallToAction.jsx

```diff
@@ -0,0 +1,65 @@
+import React, { Component } from 'react'
+import Icon from 'cozy-ui/react/Icon'
+
+import localforage from 'localforage'
+import {
+  isClientAlreadyInstalled,
+  isLinux,
+  NOVIEWER_DESKTOP_CTA
+} from 'components/pushClient'
+import styles from './styles'
+
+export default class CallToAction extends Component {
+  state = {
+    mustShow: false
+  }
+
+  async componentWillMount() {
```

Comment:
`componentWillMount` is now unsafe/deprecated, we will need to change this when we move to React (https://reactjs.org/docs/react-component.html#unsafe_componentwillmount ). In this case though, I don't really see how we can do it nicely. Maybe we can use `componentDidMount` but since we already have performance issues due to multiple rendering, it may be a bad idea.

Reply:
I edited the code, but I don't think it was the responsibility of this PR to do it. I already made an effort to separate the component in order to be able to edit it later so... ;)